src/types.ts

```diff
@@ -66,7 +66,7 @@ export interface CategoryPreferences {
   functional?: boolean | null | undefined
   marketingAndAnalytics?: boolean | null | undefined
   advertising?: boolean | null | undefined
-  [category: string]: boolean | null | undefined
+  [category: string]: boolean | null | undefined | string
```

Comment:
Is the only string allowed `N/A`? If so, I think we can replace `| string` with `| 'N/A'`. Otherwise might still be helpful to add it as an option so anyone at the type sees that it is a special value.